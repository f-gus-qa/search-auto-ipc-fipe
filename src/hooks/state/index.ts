export { default as useAutoState } from './useAutoState';
export { default as useLoadingState } from './useLoadingState';
export { default as useSelectState } from './useSelectState';
export { default as useAlertState } from './useAlertState';
