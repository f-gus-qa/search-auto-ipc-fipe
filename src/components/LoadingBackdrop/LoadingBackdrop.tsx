import { PropsWithChildren } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { CircularProgress } from '@material-ui/core';
import { useLoadingState } from '@/hooks/state';

const useStyles = makeStyles({
  fieldSet: {
    position: 'relative',
    border: 'none',
    padding: 0,
    margin: 0,
    paddingInline: 0,
    paddingBlock: 0,
    marginInline: 0,
    minInlineSize: 0,
    '*': 'inherit',
  },
  loadingContainer: {
    zIndex: 999999,
    position: 'absolute',
    width: '100%',
    height: '100%',
    backgroundColor: 'E1E1E1',
    opacity: 0.7,
    justifyContent: 'center',
    alignItems: 'center',
    display: (isLoading) => (isLoading ? 'flex' : 'none'),
  },
});

function LoadingBackdrop({ children }: PropsWithChildren<{}>) {
  const { all: isLoading } = useLoadingState();
  const classes = useStyles(isLoading);

  return (
    <fieldset data-testid="fieldset" className={classes.fieldSet} disabled={isLoading}>
      <div data-testid="containerLoading" className={classes.loadingContainer} hidden={isLoading}>
        <CircularProgress color="primary" />
      </div>
      <div>
        {children}
      </div>
    </fieldset>
  );
}

export default LoadingBackdrop;
