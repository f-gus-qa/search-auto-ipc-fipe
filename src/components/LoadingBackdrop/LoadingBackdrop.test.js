import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import LoadingBackdrop from '.';
import { useLoadingState } from '@/hooks/state';
import { fixtureLoadingAll } from '@/fixture/loading';

jest.mock('@/hooks/state', () => ({
  useLoadingState: jest.fn(),
}));

const chldren = <div>some-children</div>;
const component = <LoadingBackdrop>{chldren}</LoadingBackdrop>;

describe('<LoadingBackdrop>', () => {
  describe('when is Loading all is true', () => {
    beforeEach(() => {
      useLoadingState.mockReturnValue(fixtureLoadingAll(true));
    });

    it('check renders structure', () => {
      const { toJSON } = create(component);

      expect(toJSON()).toMatchSnapshot();
    });

    it('check container loading progress is visible', () => {
      render(component);

      const container = screen.getByTestId('containerLoading');

      expect(container).toHaveAttribute('hidden');
    });

    it('check fieldset has been disabled', () => {
      render(component);

      const fieldset = screen.getByTestId('fieldset');
      expect(fieldset).toHaveAttribute('disabled');
    });
  });

  describe('when is Loading all is false', () => {
    beforeEach(() => {
      useLoadingState.mockReturnValue(fixtureLoadingAll(false));
    });

    it('check renders structure', () => {
      const { toJSON } = create(component);

      expect(toJSON()).toMatchSnapshot();
    });

    it('check container loading progress not is visible', () => {
      render(component);

      const container = screen.getByTestId('containerLoading');

      expect(container).not.toBeVisible();
    });

    it('check fieldset has been able', () => {
      render(component);

      const fieldset = screen.getByTestId('fieldset');

      expect(fieldset.getAttribute('disabled')).toBe(null);
    });
  });
});
