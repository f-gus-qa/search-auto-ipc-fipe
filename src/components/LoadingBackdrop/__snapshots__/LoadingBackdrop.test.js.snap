// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingBackdrop> when is Loading all is false check renders structure 1`] = `
<fieldset
  className="makeStyles-fieldSet-1"
  data-testid="fieldset"
  disabled={false}
>
  <div
    className="makeStyles-loadingContainer-2 makeStyles-loadingContainer-6"
    data-testid="containerLoading"
    hidden={false}
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div>
    <div>
      some-children
    </div>
  </div>
</fieldset>
`;

exports[`<LoadingBackdrop> when is Loading all is true check renders structure 1`] = `
<fieldset
  className="makeStyles-fieldSet-1"
  data-testid="fieldset"
  disabled={true}
>
  <div
    className="makeStyles-loadingContainer-2 makeStyles-loadingContainer-3"
    data-testid="containerLoading"
    hidden={true}
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div>
    <div>
      some-children
    </div>
  </div>
</fieldset>
`;
